src/events_watcher/substrate/signature_bridge_watcher.rs

```diff
@@ -0,0 +1,401 @@
+    ) -> anyhow::Result<()> {
+        // todo
+        // if the ownership is transferred to the new owner, we need to
+        // to check our txqueue and remove any pending tx that was trying to
+        // do this transfer.
+        tracing::event!(
+            target: crate::probe::TARGET,
+            tracing::Level::DEBUG,
+            kind = %crate::probe::Kind::SignatureBridge,
+            call = "signature-bridge-maintainer-event",
+            msg = "Maintainer set",
+            new_maintainer = ?event.new_maintainer,
+            old_maintainer = ?event.old_maintainer,
+        );
+
+        // mark this event as processed.
+        // let events_bytes = &event.encode();
+        // store.store_event(events_bytes)?;
+
+        Ok(())
+    }
+}
+
+#[async_trait::async_trait]
+impl SubstrateBridgeWatcher for SubstrateBridgeEventWatcher {
+    #[tracing::instrument(skip_all)]
+    async fn handle_cmd(
+        &self,
+        node_name: String,
+        chain_id: U256,
+        store: Arc<Self::Store>,
+        ctx: RelayerContext,
+        api: Arc<Self::Api>,
+        cmd: BridgeCommand,
+    ) -> anyhow::Result<()> {
+        use BridgeCommand::*;
+        tracing::trace!("Got cmd {:?}", cmd);
+        match cmd {
+            ExecuteProposalWithSignature { data, signature } => {
+                self.execute_proposal_with_signature(
+                    node_name,
+                    chain_id,
+                    store,
+                    ctx,
+                    api.clone(),
+                    (data, signature),
+                )
+                .await?;
+            }
+            TransferOwnershipWithSignature {
+                public_key,
+                nonce,
+                signature,
+            } => {
+                self.transfer_ownership_with_signature(
+                    node_name,
+                    chain_id,
+                    store,
+                    &ctx,
+                    api.clone(),
+                    (public_key, nonce, signature),
+                )
+                .await?
+            }
+        };
+        Ok(())
+    }
+}
+
+impl SubstrateBridgeEventWatcher
+where
+    Self: SubstrateBridgeWatcher,
+{
+    #[tracing::instrument(skip_all)]
+    async fn execute_proposal_with_signature(
+        &self,
+        node_name: String,
+        chain_id: U256,
+        _store: Arc<<Self as SubstrateEventWatcher>::Store>,
+        ctx: RelayerContext,
+        api: Arc<<Self as SubstrateEventWatcher>::Api>,
+        (data, signature): (Vec<u8>, Vec<u8>),
+    ) -> anyhow::Result<()> {
+        let data_hex = hex::encode(&data);
+        let signature_hex = hex::encode(&signature);
+        // parse proposal call
+        let parsed_proposal_bytes = parse_call_from_proposal_data(&data);
+        let proposal_encoded_call: Call =
+            scale::Decode::decode(&mut parsed_proposal_bytes.as_slice())
+                .unwrap();
```

Comment:
Do not please use any `unwrap` instead handle the errors or use `?`.
```suggestion
        let proposal_encoded_call: Call =
            scale::Decode::decode(&mut parsed_proposal_bytes.as_slice())?;
```